Source/WebCore/html/HTMLDetailsElement.cpp

```diff
@@ -103,50 +103,53 @@ HTMLDetailsElement::~HTMLDetailsElement() = default;
 
 void HTMLDetailsElement::didAddUserAgentShadowRoot(ShadowRoot& root)
 {
-    auto summarySlot = HTMLSlotElement::create(slotTag, document());
+    auto summarySlot = HTMLSlotElement::create(slotTag, protectedDocument());
```

Comment:
Instead of keep calling protectedDocument() 3 times, declare a `RefPtr` local variable and store `document()` there.